### ISSUE TITLE
`permalink_prefix` for self-hosted matrix.to instances

### ISSUE THREAD

**Issue by a-0-dev** [vector-im/element-web#23933]
### Your use case

**TL;DR:** support self-hosted matrix.to instances as `permalink_prefix`.

**Explanation:**
The element web [customization documentation](https://github.com/vector-im/element-web/blob/develop/docs/config.md#customisation--branding) (no. 4) explains that `permalink_prefix` can be used to replace the matrix.to in share links with a different domain where some element web instance is located.

However, since it replaces matrix.to and not app.element.io, I would expect it would also work with self-hosted [matrix.to](https://github.com/matrix-org/matrix.to) instances. The URL path that is appended to `permalink_prefix` only works with element web though.

Comparison:
1. no `permalink_prefix` set: https://matrix.to/#/#room:example.com (working)
2. `permalink_prefix` set to an element web instance: https://my.element.web/#/room/#room:example.com (working)
3. `permalink_prefix` set to a self-hosted matrix.to: https://to.my.matrix.com/#/room/#room:example.com (**not** working - matrix.to requires the URL path scheme shown in 1)

I can imagine this is intended since many host element web and few their own matrix.to, but since matrix.to is not only handy but sometimes required it would be great if element would support it in one way or the other.

Potential solutions:
1. Auto-discover whether there's element web or matrix.to behind `permalink_prefix`
2. different config options (`permalink_prefix` or `matrix_to_prefix` for backwards compatibility)
3. (same path scheme for matrix.to and element web - not reasonable)
4. ...?

### Have you considered any alternatives?

currently, we use our self-hosted element web as prefix, but it hurts all users using desktop/mobile clients.

**Comment by turt2live** [comment]
Note that matrix.to is not intended to be much different than the `matrix:` URI scheme, meaning a "custom instance" isn't *technically* a real thing. 

matrix.to can be used in closed/private federations, even without connecting matrix.org (or its infrastructure) to your network.

I'm curious as to what your usecase is for self-hosting matrix.to though, as regardless of Product team input this doesn't look like something we'd be able to implement (safely) at a technical level, due to limitations within the spec.

**Comment by turt2live** [comment]
The spec [says](https://spec.matrix.org/v1.5/appendices/#matrixto-navigation) matrix.to URIs are for routing only, not to be assumed as a functional web server. In practice this means there's not really such a thing as a "custom matrix.to instance", but rather an onboarding service which is adjacent to the client. It's a similar request to supporting a custom scheme (`mycompany://` for example), which is where the technical side gets questionable.

**Comment by a-0-dev** [comment]
Sure, but especially when those "invite links" are shared via email, we cannot assume the e-mail client will handle matrix.to links in any special way - it will just open them in a browser. And if you already reached the point in an institution that users do not just click on links in e-mails with unknown and/or sketchy domains (and *.to would probably generally be considered sketchy) you don't really want to give that up...

So it would definitely make things easier for us if creating "self-hosted matrix.to links" in the Element UI was possible.

**Comment by turt2live** [comment]
The intention for that usecase is to use Element Web links instead, as implied by `permalinkPrefix` existing 😇 

Element has published custom mobile apps before to support this sort of stuff, though it goes through the sales channels rather than free support like here, if it helps.

**Comment by MadLittleMods** [comment]
Related to https://github.com/vector-im/element-web/issues/23960
